File: Python/Problems/Problems/problem2482.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem2482(Problem):
    def test(self) -> bool:
        grid = [[0, 1, 1], [1, 0, 1], [0, 0, 1]]

        expected = [[0, 0, 4], [0, 0, 4], [-2, -2, 2]]

        result = self.onesMinusZeros(grid)

        print2DArray(result)

        return expected == result

    def onesMinusZeros(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        m = len(grid[0])
        rowOnes = [0] * n
        rowZeros = [0] * n
        colOnes = [0] * m
        colZeros = [0] * m
        for i in range(n):
            for j in range(m):
                if grid[i][j] == 0:
                    rowZeros[i] += 1
                    colZeros[j] += 1
                elif grid[i][j] == 1:
                    rowOnes[i] += 1
                    colOnes[j] += 1
        result = []
        for i in range(n):
            row = []
            for j in range(m):
                row.append(rowOnes[i] + colOnes[j] - rowZeros[i] - colZeros[j])
            result.append(row)

        return result

File: C++/Problems/Problems/problem_2482.cpp
#include "problem_2482.h"

bool problem_2482::test() {
    std::vector<std::vector<int>> grid {  { 0, 1, 1 },  { 1, 0, 1 }, { 0, 0, 1 } };

    std::vector<std::vector<int>> expected  {  { 0, 0, 4 },  { 0, 0, 4 },  { -2, -2, 2 } };

    auto result = onesMinusZeros(grid);

    print2DVector(result);

    return expected == result;
}

std::vector<std::vector<int>> problem_2482::onesMinusZeros(std::vector<std::vector<int>> &grid) {
    int n = grid.size();
    int m = grid[0].size();
    std::vector rowOnes(n, 0);
    std::vector rowZeros(n, 0);
    std::vector colOnes(m, 0);
    std::vector colZeros(m, 0);
    for (int i = 0; i < n; ++i)
    {
        for (int j = 0; j < m; ++j)
        {
            if (grid[i][j] == 0)
            {
                rowZeros[i]++;
                colZeros[j]++;
            }
            else if (grid[i][j] == 1)
            {
                rowOnes[i]++;
                colOnes[j]++;
            }
        }
    }
    std::vector result(n, std::vector<int>(m));
    for (int i = 0; i < n; ++i)
    {
        for (int j = 0; j < m; ++j)
            result[i][j] = rowOnes[i] + colOnes[j] - rowZeros[i] - colZeros[j];
    }

    return result;
}
